### PR TITLE
Add an error code for invalid user names.

### PR DESCRIPTION
**Pull request by dbkr**

### CODE REVIEW

File: api/client-server/registration.yaml [hunk 1]
```diff
@@ -100,6 +100,7 @@ paths:
             Part of the request was invalid. This may include one of the following error codes:
 
             * ``M_USER_IN_USE`` : The desired user ID is already taken.
+            * ``M_INVALID_USER_NAME`` : The desired user ID is not a valid user name.
```
Comment (illicitonion) [inline api/client-server/registration.yaml]: The other USER error refers to user IDs just as USER (M_USER_IN_USE), maybe use the same to be consistent?

**Comment by illicitonion** [comment]
LGTM, one suggestion

File: api/client-server/registration.yaml (hunk 1 above)
Comment (dbkr) [inline api/client-server/registration.yaml]: Asked to about this to see what others though, suggestion seems to be M_INVALID_USERNAME to match the parameter to the API. Thoughts?

Comment (illicitonion) [inline api/client-server/registration.yaml]: Works for me :)

(I can also live with it as it is)

(Also, it's a shame we're so inconsistent with our names everywhere)

Comment (dbkr) [inline api/client-server/registration.yaml]: Yeah :/

**Comment by illicitonion** [comment]
Still LGTM :)

**Pull request merged**